mockative-test/src/wasmWasiMain/kotlin/io/mockative/concurrency/AtomicReference.kt

```diff
@@ -0,0 +1,8 @@
+package io.mockative.concurrency
+
+internal actual class AtomicReference<T> actual constructor(actual var value: T) {
+    actual fun compareAndSet(expected: T, new: T): Boolean {
+        value = new
+        return true
+    }
+}
```

Comment:
I know that this is not really the scope of the pr but this should probably be updated to be

```suggestion
internal actual class AtomicReference<T> actual constructor(actual var value: T) {
        actual fun compareAndSet(expected: T, new: T): Boolean {
        if (value == expected) {
            value = new
            return true
        }
        return false
    }
}
```

Reply:
Good point! I could probably completely remove the atomic references though... They are a remnant from the time when object freezing was a thing